src/cache_cf.cc

```diff
@@ -543,9 +553,17 @@ parseOneConfigFile(const char *file_name, unsigned int depth)
             /* Handle includes here */
             if (tmp_line_len >= 9 && strncmp(tmp_line, "include", 7) == 0 && xisspace(tmp_line[7])) {
                 err_count += parseManyConfigFiles(tmp_line + 8, depth + 1);
-            } else if (!parse_line(tmp_line)) {
-                debugs(3, DBG_CRITICAL, HERE << cfg_filename << ":" << config_lineno << " unrecognized: '" << tmp_line << "'");
-                ++err_count;
+            } else {
+                try {
+                    if (!parse_line(tmp_line)) {
+                        debugs(3, DBG_CRITICAL, ConfigPosition() << "unrecognized: '" << tmp_line << "'");
+                        ++err_count;
+                    }
+                } catch (const std::exception &ex) {
+                    // fatal for now
+                    debugs(3, DBG_CRITICAL, ex.what());
```

Comment:
We should add some prefix here in case ex.what() does not give an admin a good clue that this is a configuration error. For example:
```suggestion
                    debugs(3, DBG_CRITICAL, "configuration error: " << CurrentException);
```